s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/S3SourceTask.java

```diff
@@ -71,12 +72,12 @@ public class S3SourceTask extends SourceTask {
     private S3SourceConfig s3SourceConfig;
```

Comment:
The default graceful shutdown time is 5 seconds, so a 10 second poll interval will have a pretty high chance of causing ungraceful shutdown: https://github.com/apache/kafka/blob/5313f8eb92a033dc74d8e72a48ac033113b186d4/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java#L101

You need to ensure that the poll() method returns multiple times while you're waiting for back-offs for other operations.

Reply:
Will be handled in ticket https://aiven.atlassian.net/browse/KCON-26